File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max = Integer.MIN_VALUE;
        int maxTillNow = 0;
        for(int i=0; i<nums.length; i++){
            maxTillNow = Math.max(nums[i], maxTillNow+nums[i]);
            max = Math.max(max, maxTillNow);
        }
        return max;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxi = nums[0];
        int maxPosstillprev = nums[0];
        for(int i =1; i<nums.size(); i++){
            maxPosstillprev =  max(nums[i], nums[i]+maxPosstillprev);
            maxi = max(maxi, maxPosstillprev);
        }
        return maxi;
    }
};